app/components/shared/input/template.hbs

```diff
@@ -0,0 +1,8 @@
+<input
+  type={{this.chceckInputType}}
+  class='form-control {{@class}} {{this.size}}'
+  id={{@id}}
+  placeholder='{{@placeholder}}'
+  value={{@vale}}
+  {{on 'input' @onChange}}
+/>
```

Comment:
Add `...attributes` then you can remove Ember `@class` attribute, then you can pass natively `class` prop, then will be concat other classes to this element (similar like you have in shared/button):
```suggestion
<input
  type={{this.chceckInputType}}
  class='form-control {{this.size}}'
  id={{@id}}
  placeholder='{{@placeholder}}'
  value={{@vale}}
  {{on 'input' @onChange}}
  ...attributes
/>
```